statsmodels/graphics/tsaplots.py

```diff
@@ -876,3 +877,73 @@ def plot_predict(
     ax.legend(loc="best")
 
     return fig
+
+
+def seasonal_diagnostic_plot(x, period_length, select=None, 
+                             subseries_labels=None, nrows=1, **kwargs):
+    """
+    Seasonal-Diagnostic Plot, as described by [1]_.
+
+
+    Parameters
+    ----------
+    x : DecomposeResult
+        The result of your seasonal decomposition.
+    period_length : int
+        The length of the period. Should match the `period` parameter used to
+        decompose the series.
+    select : int
+        The number of periods to plot. The periods are selected evenly from
+        `range(period_length)`. By default, all periods are displayed.
+    subseries_labels : array of str
+        Labels for the displayed period subplots.
+    nrows : int
+        The number of rows on which to display the plots.
+
+
+    Returns
+    -------
+    fig : Figure
+        Returns an matplotlib object of type Figure with the
+        Seasonal-Diagnostic Plot.
+
+
```

Comment:
Perhaps add an 

```
Example
-------
```

section. Ideally one that will generate an actual plot which can then appear in the web docs.